### DISCUSSION TITLE
Error building docker image of coffeEditor

### DISCUSSION THREAD

**Discussion by jha659** [eclipse-emfcloud/emfcloud#213] [Coffee Editor] [answered]
Hi, 

I am following repository instructions to build a docker image of CoffeEditor, but, when I try to execute this command:
`docker build -t "coffee-editor:latest" .`

all start building (client and server) but in some point of the execution it fails, this is the info showed on console:

`#21 101.6 coffee-welcome-page: src/browser/welcome-page-widget.tsx(23,33): error TS2307: Cannot find module 'coffee-workflow-analyzer/lib/browser/command-contribution' or its corresponding type declarations.
#21 101.7 coffee-welcome-page: error Command failed with exit code 2.
#21 101.7 coffee-welcome-page: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
#21 101.8 coffee-welcome-page: error Command failed with exit code 2.
#21 101.8 coffee-welcome-page: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
#21 101.8 lerna ERR! yarn run prepare exited 2 in 'coffee-welcome-page'
#21 101.8 lerna WARN complete Waiting for 2 child processes to exit. CTRL-C to exit immediately.
#21 101.9 error Command failed with exit code 2.
#21 101.9 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
------
executor failed running [/bin/sh -c yarn install &&     yarn production]: exit code: 2`

I am on a Macbook Air M1, could anyone help me please? thanks in advance!

**Comment by jha659** [comment]
UPDATE:

I have updated my Docker version to 4.20.1 and now building it throws this error:

`#0 78.63 Setting up ca-certificates-java (20190909) ...
#0 78.67 head: cannot open '/etc/ssl/certs/java/cacerts' for reading: No such file or directory
#0 78.84 Exception in thread "main" java.lang.InternalError: Error loading java.security file
#0 78.84 	at java.base/java.security.Security.initialize(Security.java:106)
#0 78.84 	at java.base/java.security.Security$1.run(Security.java:84)
#0 78.84 	at java.base/java.security.Security$1.run(Security.java:82)
#0 78.84 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
#0 78.84 	at java.base/java.security.Security.<clinit>(Security.java:82)
#0 78.84 	at java.base/sun.security.jca.ProviderList.<init>(ProviderList.java:178)
#0 78.84 	at java.base/sun.security.jca.ProviderList$2.run(ProviderList.java:96)
#0 78.84 	at java.base/sun.security.jca.ProviderList$2.run(ProviderList.java:94)
#0 78.84 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
#0 78.84 	at java.base/sun.security.jca.ProviderList.fromSecurityProperties(ProviderList.java:93)
#0 78.84 	at java.base/sun.security.jca.Providers.<clinit>(Providers.java:55)
#0 78.84 	at java.base/sun.security.jca.GetInstance.getInstance(GetInstance.java:156)
#0 78.84 	at java.base/java.security.cert.CertificateFactory.getInstance(CertificateFactory.java:193)
#0 78.84 	at org.debian.security.KeyStoreHandler.<init>(KeyStoreHandler.java:50)
#0 78.84 	at org.debian.security.UpdateCertificates.<init>(UpdateCertificates.java:65)
#0 78.84 	at org.debian.security.UpdateCertificates.main(UpdateCertificates.java:51)
#0 78.85 dpkg: error processing package ca-certificates-java (--configure):
#0 78.85  installed ca-certificates-java package post-installation script subprocess returned error exit status 1
#0 78.85 dpkg: dependency problems prevent configuration of openjdk-17-jre-headless:arm64:
#0 78.85  openjdk-17-jre-headless:arm64 depends on ca-certificates-java (>= 20190405~); however:
#0 78.85   Package ca-certificates-java is not configured yet.
#0 78.85
#0 78.85 dpkg: error processing package openjdk-17-jre-headless:arm64 (--configure):
#0 78.85  dependency problems - leaving unconfigured
#0 78.85 dpkg: dependency problems prevent configuration of openjdk-17-jre:arm64:
#0 78.85  openjdk-17-jre:arm64 depends on openjdk-17-jre-headless (= 17.0.7+7-1~deb11u1); however:
#0 78.85   Package openjdk-17-jre-headless:arm64 is not configured yet.
#0 78.85
#0 78.85 dpkg: error processing package openjdk-17-jre:arm64 (--configure):
#0 78.85  dependency problems - leaving unconfigured
#0 78.85 dpkg: dependency problems prevent configuration of openjdk-17-jdk:arm64:
#0 78.85  openjdk-17-jdk:arm64 depends on openjdk-17-jre (= 17.0.7+7-1~deb11u1); however:
#0 78.85   Package openjdk-17-jre:arm64 is not configured yet.
#0 78.85
#0 78.85 dpkg: error processing package openjdk-17-jdk:arm64 (--configure):
#0 78.85  dependency problems - leaving unconfigured
#0 78.85 dpkg: dependency problems prevent configuration of openjdk-17-jdk-headless:arm64:
#0 78.85  openjdk-17-jdk-headless:arm64 depends on openjdk-17-jre-headless (= 17.0.7+7-1~deb11u1); however:
#0 78.85   Package openjdk-17-jre-headless:arm64 is not configured yet.
#0 78.85
#0 78.85 dpkg: error processing package openjdk-17-jdk-headless:arm64 (--configure):
#0 78.85  dependency problems - leaving unconfigured
#0 78.85 Processing triggers for libc-bin (2.31-13+deb11u6) ...
#0 78.96 Processing triggers for ca-certificates (20210119) ...
#0 78.97 Updating certificates in /etc/ssl/certs...
#0 79.81 0 added, 0 removed; done.
#0 79.81 Running hooks in /etc/ca-certificates/update.d...
#0 79.81
#0 79.84 /etc/ca-certificates/update.d/jks-keystore: 82: java: not found
#0 79.84 E: /etc/ca-certificates/update.d/jks-keystore exited with code 1.
#0 79.84 done.
#0 79.92 Errors were encountered while processing:
#0 79.92  ca-certificates-java
#0 79.92  openjdk-17-jre-headless:arm64
#0 79.92  openjdk-17-jre:arm64
#0 79.92  openjdk-17-jdk:arm64
#0 79.92  openjdk-17-jdk-headless:arm64
#0 80.43 E: Sub-process /usr/bin/dpkg returned an error code (1)
------
Dockerfile:59
--------------------
  58 |     # Theia dependencies/Java
  59 | >>> RUN apt-get update &&\
  60 | >>>     apt-get -y install --no-install-recommends \
  61 | >>>     software-properties-common \
  62 | >>>     libxkbfile-dev \
  63 | >>>     libsecret-1-dev openjdk-17-jdk \
  64 | >>>     build-essential libssl-dev wget gnupg git gdb
  65 |
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update &&    apt-get -y install --no-install-recommends     software-properties-common     libxkbfile-dev     libsecret-1-dev openjdk-17-jdk     build-essential libssl-dev wget gnupg git gdb" did not complete successfully: exit code: 100
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.`

Any suggestion to solve it? thanks in advance!

**Reply by ndoschek** [reply, marked as answer]
Hi @jha659!

Thanks for catching that! Apparently there were changes made in OpenJDK that require now to install the `ca-certificates-java` package to install `openjdk-17-jdk` properly.
We updated the Dockerfile accordingly, with the latest master of the `coffee-editor` the docker build should work again as expected!
See also https://github.com/eclipse-emfcloud/coffee-editor/pull/510

Best regards,
Nina

**Reply by jha659** [reply]
hi, thanks! now it works on windows 10 x64, but it is still failing on macbook with M1 processor (it is weird), anyway I can try it on windows, thanks a lot!
